packages/url/src/index.js

```diff
@@ -15,11 +16,10 @@ const USABLE_HREF_REGEXP = /^(?:[a-z]+:|#|\?|\.|\/)/i;
  * @return {string}       Updated URL
  */
 export function addQueryArgs( url, args ) {
-	const parsedURL = parse( url, true );
-	const query = { ...parsedURL.query, ...args };
-	delete parsedURL.search;
+	const parsedURL = parseURL( url );
+	const query = { ...parse( parsedURL.query ), ...args };
 
-	return format( { ...parsedURL, query } );
+	return stringifyURL( { ...parsedURL, search: stringify( query ) } );
```

Comment:
I'd be motivated to simplify this function further so we can remove the dependency on `url` (as mentioned previously, it's quite large).

I think this should work okay?

```js
const search = stringify( args );
const joiner = url.indexOf( '?' ) === -1 ? '?' : '&';
return url + joiner + search;
```

Reply:
I think there's a difference between the two implementation. In this particular case, we don't overwrite already present args in the url query string, but I think I can update the implementation to avoid the url package and still override existing keys.